server/embed/etcd.go

```diff
@@ -774,3 +797,29 @@ func parseCompactionRetention(mode, retention string) (ret time.Duration, err er
 	}
 	return ret, nil
 }
+
+func (e *Etcd) setupTracing(ctx context.Context) (tracesdk.SpanExporter, error) {
+	exporter, err := otlp.NewExporter(ctx,
+		otlpgrpc.NewDriver(
+			otlpgrpc.WithEndpoint(e.cfg.ExperimentalDistributedTracingAddress),
+			otlpgrpc.WithInsecure(),
+		))
+	if err != nil {
+		return nil, err
+	}
+	res := resource.NewWithAttributes(
+		semconv.ServiceNameKey.String(e.cfg.ExperimentalDistributedTracingServiceName),
+	)
+	tp := tracesdk.NewTracerProvider(
+		tracesdk.WithSyncer(exporter),
+		tracesdk.WithResource(res),
+	)
+	otel.SetTracerProvider(tp)
```

Comment:
Just to call this out to reviewers less familiar with opentelemetry: This is setting a global variable, which is used by tracing libraries.  In the kubernetes APIServer, we will likely avoid doing this by explicitly passing the TracerProvider to the libraries we want to do tracing.  It is more code, but avoids global variables.

Reply:
Thanks! Yeah right now we are not adding any spans only using the gRPC so I don't see it being different, but I could be wrong?

Reply:
If, for example, a library you rely on added OpenTelemetry tracing, you would start getting traces from that library without changing anything on your end, since it will look up the global TracerProvider to see where to send traces.  That may or may not be desirable.  Also, if you have multiple different "services" in the same go binary, you can't customize the TracerProviders individually for each service while using the global TracerProvider.  I think it is perfectly fine to start with using the global one, though.

Reply:
Embed server is being used 'as a library' by our customers. Also in tests we create multiple instances of Embed server to mimic multiple etcd nodes. Recently we get rid of 'global loggers'  (including grpc loggers) being overridden by each of the embed servers (leading to misleading zap-fields being reported). 

I don't know how differently `otl` can be configured for different libs, but  having lib-specific tracer seems to be on the safer side.

Reply:
I am all for removing globals, I felt the pain many times with Prometheus global registry (which I would love to remove from etcd one day as well, if folks agree). 

From what I can tell we can pass it via https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc#WithTracerProvider, so will do that.

Reply:
Agreed.  I opened https://github.com/open-telemetry/opentelemetry-go/issues/1888

Reply:
> Embed server is being used 'as a library' by our customers. Also in tests we create multiple instances of Embed server to mimic multiple etcd nodes.

If the embedded server is used as a library it may be better to avoid use of the OTel SDK at all and only use the API.  In that case the server should receive its TracerProvider and propagators using the API interfaces from the `main` package that has instantiated them with the SDK.  That would allow the consumer of the embedded server to decide whether to enable tracing, which exporters and propagators to use, even which SDK to use if alternate implementations of the SDK become available.